### ISSUE TITLE
Tasks only displays the first category

### ISSUE THREAD

**Issue by heiko77** [nextcloud/tasks#1102]
### Steps to reproduce
1. Create tasks with multiple categories in Outlook and sync the task with CalDav Synchronizer for Outlook
2. Task plugin show only the first categories

### Expected behaviour
CalDav Synchronizer send categories in multiple CATEGORIES properties and task show all categories.
```
...
BEGIN:VTODO
CATEGORIES:red category
CATEGORIES:blue category
CATEGORIES:green category
CATEGORIES:yellow category
..
END:VTODO
```

### Actual behaviour
Only the first category "red category" is displayed
Task plugin only display multiple VTODOS categories in following format:
```
...
BEGIN:VTODO
CATEGORIES:red category,blue category,green category,yellow category
..
END:VTODO
```

**Comment by raimund-schluessler** [comment]
I am not sure it is even allowed to specify the `CATEGORIES` property multiple times in a `VTODO`. RFC5545 https://tools.ietf.org/html/rfc5545#section-3.8.1.2 does not explicitly specify how often this property can appear (where e.g. for `ATTACH` it states it is allowed multiple times). What it does state is, that multiple categories can be separated by a comma: 

> Within the "VEVENT", "VTODO", or "VJOURNAL" calendar components, more than one category can be specified as a COMMA-separated list of categories.

So my take on this is that the `VTODO` produces by CalDav Synchronizer is actually not valid and multiple categories should be stored as a comma separated list. @georgehrke What do you think about that?

**Comment by georgehrke** [comment]
According to the definition of `VTODO`, the `CATEGORIES` property is allowed to occur more than once:
https://tools.ietf.org/html/rfc5545#section-3.6.2

>                   ; The following are OPTIONAL,
>                   ; and MAY occur more than once.
>                   ;
>                  attach / attendee / categories / comment / contact /
>                  exdate / rstatus / related / resources /
>                  rdate / x-prop / iana-prop
>                  ;

I'm also supporting it that way in the Calendar app. By default, it will put them all into one `CATEGORIES` property, but it is able to handle multiple ones.

**Comment by raimund-schluessler** [comment]
Alright, thanks a lot for the clarification. We should adjust it in the Tasks app then.

**Comment by raimund-schluessler** [comment]
@georgehrke Could you point me to the code where you handle categories in Calendar? I try to figure out how to remove a single category (value) from the correct `CATEGORIES` property.

**Comment by georgehrke** [comment]
In the calendar app that is all handled by calendar.js:
https://github.com/georgehrke/calendar-js/blob/c707a64db0c3ef57de0eb4be43bea9d2345d67c8/src/components/abstractComponent.js#L669

**Comment by raimund-schluessler** [comment]
Hm, maybe I should start using calendar-js as well for Tasks 🤔

**Comment by raimund-schluessler** [comment]
Fix is in #1114.
